src/WebJobs.Script.WebHost/Security/KeyManagement/BlobStorageSecretsMigrationRepository.cs

```diff
@@ -0,0 +1,158 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using Microsoft.WindowsAzure.Storage.Blob;
+
+namespace Microsoft.Azure.WebJobs.Script.WebHost
+{
+    public class BlobStorageSecretsMigrationRepository : ISecretsRepository, IDisposable
+    {
+        private readonly string _secretSentinelDirectoryPath;
+        private readonly string _accountConnectionString;
+        private readonly string _siteSlotName;
+        private readonly ILogger _logger;
+        private Lazy<BlobStorageSecretsRepository> _blobStorageSecretsRepository;
+
+        public BlobStorageSecretsMigrationRepository(string secretSentinelDirectoryPath, string accountConnectionString, string siteSlotName, ILogger logger)
+        {
+            _secretSentinelDirectoryPath = secretSentinelDirectoryPath;
+            _accountConnectionString = accountConnectionString;
+            _siteSlotName = siteSlotName;
+            _logger = logger;
+        }
+
+        public event EventHandler<SecretsChangedEventArgs> SecretsChanged;
+
+        private BlobStorageSecretsRepository BlobStorageSecretsRepository
+        {
+            get
+            {
+                if (_blobStorageSecretsRepository == null)
```

Comment:
You can simplify this by instantiating `_blobStorageSecretsRepository` at construction and making it `readonly`. Repository instantiation would still be lazily done.